src/functionaltest/java/com/ericsson/ei/frontend/SubscriptionHandlingFunctionality.java

```diff
@@ -0,0 +1,138 @@
+
+import org.junit.*;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+import com.ericsson.ei.config.SeleniumConfig;
+import com.ericsson.ei.frontend.pageobjects.IndexPage;
+import com.ericsson.ei.frontend.pageobjects.SubscriptionPage;
+
+public class SubscriptionHandlingFunctionality extends SeleniumBaseClass {
+
+    private static final String DOWNLOADEDTEMPLATEFILEPATH = String.join(File.separator,
+            SeleniumConfig.getTempDownloadDirectory().getPath(), "subscriptionsTemplate.json");
+    private static final String SUBSCRIPTIONTEMPLATEFILEPATH = String.join(File.separator, "src", "functionaltest",
+            "resources", "responses", "SubscriptionTemplate.json");
+    private static final String RELOADTESTFILEPATH = String.join(File.separator, "src", "functionaltest", "resources",
+            "responses", "SubscriptionForUploadCase.json");
+    private static final String SAVETESTFILEPATH = String.join(File.separator, "src", "functionaltest", "resources",
+            "responses", "SubscriptionForSaveCase.json");
+    private static final String UPLOADFILEPATH = String.join(File.separator, "src", "functionaltest", "resources",
+            "responses", "SubscriptionForUploadCase.json");
+
+    @Test
+    public void testSubscription() throws Exception {
+
+        // Open index page
+        IndexPage indexPageObject = new IndexPage(mockedHttpClient, driver, baseUrl);
+        indexPageObject.loadPage();
+
+        // Click on Subscription Handling page button and verify that it is open
+        String headerPath = "//div[@class='container pull-left']//h1";
+        SubscriptionPage subscriptionPage = indexPageObject.clickSubscriptionPage();
+        new WebDriverWait(driver, 10).until((webdriver) -> subscriptionPage.presenceOfHeader(headerPath));
+        
+        // // Press "Reload" button and verify that two subscriptions with names "Subscription1" and "Subscription2" are present
+         String response = this.getJSONStringFromFile(RELOADTESTFILEPATH);
+         subscriptionPage.clickReload(response);
+         assert (driver.getPageSource().contains("Subscription1"));
+         assert (driver.getPageSource().contains("Subscription2"));
+        
+        // Click "Add Subscription" button and verify that "Subscription Form" is open
+         subscriptionPage.clickAddSubscription();
+         String xPath = "//*[@id='formHeader']";
+         new WebDriverWait(driver, 10).until((webdriver) -> (subscriptionPage.presenceOfHeader(xPath)));
+    
+         // On subscription form, select the template as "Mail Trigger" and verify
+         String tempPath = "//select[contains(@title,'Choose a Subscription Template')]";
+         String tempMail = "Mail Trigger";
+         subscriptionPage.selectDropdown(tempPath, tempMail);
+         TimeUnit.SECONDS.sleep(2);
```

Comment:
Do you really need a sleep here? If you are waiting for an item in the dropdown to appear, there is a function called 
ExpectedConditions.elementToBeClickable(By.id)

I think its the same with the rest sleeps. They should be possible to be replaced by ExpectedConditions